var/spack/repos/builtin/packages/vecmem/package.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Vecmem(CMakePackage):
+    """VecMem is a vectorised data model base and helper classes."""
+
+    homepage = "https://github.com/acts-project/vecmem"
+    url      = "https://github.com/acts-project/vecmem/archive/refs/tags/v0.5.0.tar.gz"
+    list_url = "https://github.com/acts-project/vecmem/releases"
+
+    maintainers = ['wdconinc', 'HadrienG2']
+
+    version('0.8.0', sha256='a13f7178c940d6bf3386e7e8f5eb158e6435882533bffe888d3c9775eeb2f20e')
+    version('0.7.0', sha256='c00266bc80df8f568103f2874ce349fe8f74fb7e361901d562cce41ab7f9b85c')
+    version('0.6.0', sha256='e6c8262844a5ff7f03df7f849a1e7cf1a68214730ac54c35c14333522ff31723')
+    version('0.5.0', sha256='b9739e8fcdf27fa9ef509743cd8f8f62f871b53b0a63b93f24ea9865c2b00a3a')
+    version('0.4.0', sha256='51dfadc2b97f34530c642abdf86dcb6392e753dd68ef011bac89382dcf8aaad4')
+    version('0.3.0', sha256='4e7851ab46fee925800405c5ae18e99b62644d624d3544277a522a06fb812dbf')
+    version('0.2.0', sha256='33aea135989684e325cb097e455ff0f9d1a9e85ff32f671e3b3ed6cc036176ac')
+    version('0.1.0', sha256='19e24e3262aa113cd4242e7b94e2de34a4b362e78553730a358f64351c6a0a01')
+
+    variant('cuda', default=False, description='Build the vecmem::cuda library')
+    variant('hip', default=False, description='Build the vecmem::hip library')
+    variant('sycl', default=False, description='Build the vecmem::sycl library')
+
+    depends_on('cmake@3.8:', type='build')
+    depends_on('cuda', when='+cuda')
+    depends_on('hip', when='+hip')
+    depends_on('sycl', when='+sycl')
```

Comment:
I'm a bit surprised about the low version number required from CMake. 3.8 would not be able to build even just the core library of the package.

https://github.com/acts-project/vecmem/blob/main/CMakeLists.txt#L8

But with CUDA enabled, the CMake requirement increases even further. (https://github.com/acts-project/vecmem/blob/main/cuda/CMakeLists.txt#L8)

I'm not convinced that spack should code that much information about the project's CMake dependence into its build recipe. But the minimum requirement should really be set to a very minimum of 3.10. Or if you just want to make sure that every part of vecmem can be built, to 3.17.

Reply:
> I'm not convinced that spack should code that much information about the project's CMake dependence into its build recipe. But the minimum requirement should really be set to a very minimum of 3.10. Or if you just want to make sure that every part of vecmem can be built, to 3.17.

Spack needs to, because it is a full-stack package manager that will install (almost) every dependency including cmake for you. Solving the dependency graph requires knowing which version of each dependency is suitable for installation.

Reply:
I just meant that spack should not need to know that:
  - the absolute minimum is 3.10;
  - if you turn on testing, the minimum becomes 3.11;
  - if you turn on CUDA, the minimum becomes 3.17.

I'm on board with it having one minimum. Which I guess should be 3.17, to be on the safe side...

Reply:
Ah ok, I thought you meant spack should not know about cmake version requirements. Then we agree. I would also just specify `cmake@3.17:` because it's usually enough and people can still add the full story later on if they ever need a package that only builds with old cmake (which has not happened yet thankfully).

Reply:
Implemented.